src/main/java/gift/entity/Product.java

```diff
@@ -0,0 +1,68 @@
+package gift.entity;
+
+// 관리할 객체 만들기
+public class Product {
+    private long id;
+    private String name;
+    private int price;
+    private String imgUrl;
+
+    public Product(Long o, String name, int price, String imgUrl) {}
+
+    // id 포함하지 않고 필드 초기화
+    public Product(String name, int price, String imgUrl) {
+        this.name = name;
+        this.price = price;
+        this.imgUrl = imgUrl;
+    }
+
+    // id 포함해서 필드 초기화
+    public Product(long id, String name, int price, String imgUrl) {
+        this.id = id;
+        this.name = name;
+        this.price = price;
+        this.imgUrl = imgUrl;
+    }
```

Comment:
method chaining을 이용하시면 더 간결하게 표현할 수 있습니다.

```suggestion
    // id 포함하지 않고 필드 초기화
    public Product(String name, int price, String imgUrl) {
        this(null, name, price, imgUrl);
    }

    // id 포함해서 필드 초기화
    public Product(long id, String name, int price, String imgUrl) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.imgUrl = imgUrl;
    }
```